addons/udes_stock/models/stock_move_line.py

```diff
@@ -332,6 +332,84 @@ def prepare(
                 quantity_fulfilled = 0
                 if is_serial:
                     qty_done = 1
+
+                # Swapping the tracked products
+                lot_names_on_picking = self.lot_id.mapped("name")
```

Comment:
Can we refactor this out into its own method? `prepare` is already huge and is fast becoming the new `update_picking` from v11!

Maybe we can do something like:

`lot_names = self._get_lot_names`

Reply:
I couldn't see a nice way to do this as there will be lots of arguments going into the function and more than one coming out, but I the prepare() method is too long, so I will do it